### DISCUSSION TITLE
Performance Settings (Sample Rate and Buffer Size)

### DISCUSSION THREAD

**Discussion by kleinfa** [robert-rc2i/ptq-client-webapp#54] [Q&A] [answered]
Hi! First of all: Great work, extremely useful. Thank you so much for this. Now, on to my question: I use Pianoteq on a headless Raspberry Pi, and it sometimes seems to struggle a little bit CPU-wise. It would be great if we could use the web interface to change performance options, too, or at least get some indication on the current performance index or latency. I assume Pianoteq doesn't share this information, or else you would have included that already in the web interface. Or is there maybe a way we could add this? Feedback would be greatly appreciated! Many thanks again!

**Comment by robert-rc2i** [comment, marked as answer]
Hi,

First, thanks for the kind words about this project.  

As for your request, I am afraid that it is not possible with the current version of Pianoteq's API.  The parameters that we can change through the JSON RPC API,  that Pianoteq makes available, does not include the audio interfaces configuration.  I can list all possible audio interfaces and the one being selected, but I cannot change its config through the available API nor even set a different audio interface from the list of available ones.

Sorry, but it is not possible right now.  I'll keep an eye on this with future release of Pianoteq.  Also, I suggest that you open a feature request to Modartt, to let them know about your need.  Never know!

Cheers,

**Reply by kleinfa** [reply]
Hi! Thank you very much for the informative answer. I was afraid this would be your answer. However, I will do as you suggested and request the feature with Modartt. Anyway, thanks again and all the best!
